[0030]  daemon@ATHENA.MIT.EDU (Peter MacDonald) Linux_Activists 11/10/91 20:34 (39 lines)
Subject: shoelace patch #2
Date: Sun, 10 Nov 91 16:57:48 PST
From: pmacdona@sol.UVic.CA (Peter MacDonald)
To: linux-activists@joker.cs.hut.fi


There is a big difference between floppy the device numbers on linux and
on minix.  In particular linux has fd0 as 0x208 while minix uses 0x200.
ie.  linux /dev/fd0 == minix /dev/fd0-dshd5in

The following patch will keep linux from using 0x200, which it doesn't 
know about and instead use 0x208 and 0x21c.
Apply this after all other patches.

pmacdona@sol.uvic.ca



*** shoe.c.bad	Thu Nov  7 08:43:30 1991
--- shoe.c	Thu Nov  7 08:52:27 1991
***************
*** 638,646 ****
        longjmp(errjmp, 1);*/
        unsigned *myptr = (unsigned int *) (&((char *) bp)[508]);
        char *myroot = getenvs(SaveConfigPtr,"rootdev");
!       if (strcmp(myroot,"bootdev"))
!         *myptr = atoi(myroot);
!       else
          *myptr = 0;
        if (*myptr == 0)
        { if (diskcode)
--- 638,645 ----
        longjmp(errjmp, 1);*/
        unsigned *myptr = (unsigned int *) (&((char *) bp)[508]);
        char *myroot = getenvs(SaveConfigPtr,"rootdev");
!       *myptr = atoi(myroot);
!       if (0x200 == *myptr)
          *myptr = 0;
        if (*myptr == 0)
        { if (diskcode)
--[0030]--
